pyb4model/pyb4model.py

```diff
@@ -61,50 +66,144 @@ def fit_and_report(model, X, y, Xv, yv, m_type = 'regression'):
-        fs = ForSelect(rf, min_features=2, max_features=5)
-        fs.fit(X_train, y_train)
-        fs.ftr_
+        selected_features = ForSelect(rf, 
+                                    X_train, 
+                                    y_train,
+                                    min_features=2, 
+                                    max_features=5, 
+                                    scoring="neg_mean_square",
+                                    problem_type="regression", 
+                                    cv=4)
+        new_X_train = X_train[selected_features]
         """
 
+        # Test the input & arguments
+        Test_ForSelect()
+
+        # Create Empty Feature list
+        ftr_ = []
+
+        # Define maximum amount of features
+        if max_features is None:
+            max_features = X.shape[1]
+
+        # total list of features
+        total_ftr = list(range(0, X.shape[1]))
+
+        # define scoring
+        if problem_type=="regression":
+            scoring='neg_mean_squared_error',
+        else:
+            scoring='accuracy'
+        
+        # initialize error score
+        best_score = -np.inf
+
+        i = 0
+
+        while len(ftr_) < max_features:
+            # remove already selected features
+            features_unselected = list(set(total_ftr) - set(ftr_))
+
+            # Initialize potential candidate feature to select
+            candidate = None
+
+            # Iterate
+            for feature in features_unselected:
+                ftr_candidate = ftr_ + [feature]
+                eval_score = np.mean(cross_val_score(model,
+                                                     X[:, ftr_candidate],
+                                                     y,
+                                                     cv=cv,
+                                                     scoring=scoring))
+
+                # If computed error score is better than our current best score
+                if eval_score > best_score:
+                    best_score = eval_score  # Overwrite the best_score
+                    candidate = feature  # Consider the feature as candidate
+
+            # Add the selected feature
+            if candidate is not None:
+                ftr_.append(candidate)
+
+                # Report Progress
+                i = i + 1
+                if i % 5 == 0:
+                    print("{} Iterations Done".format(i))
+                    print("current best score: {}".format(best_score))
+
+                    print("Current selected features: {}".format(ftr_))
+                    print("\n")
+
+            else:
+                # End process
+                print("{} iterations in total".format(i))
+                print("Final selected features: {}".format(ftr_))
+                break
+
+        # End Process
+        print("{} iterations in total".format(i))
+        print("Final selected features: {}".format(ftr_))
+
+        return ftr_
+
+
+def Test_ForSelect():
```

Comment:
- create test function using pytest, do not add test function in this file
- see tiffany's tutorial: https://ubc-mds.github.io/py-pkgs/whole-game.html#poetry-testing